# 💿 Spotify Tracks
## 🎵 Hypothesis 3 — Acousticness vs Energy & Popularity

| Field         | Description |
|---------------|-------------|
| Author:       | Robert Steven Elliott |
| Course:       | Code Institute – Data Analytics with AI Bootcamp |
| Project Type: |	Hackathon 2 |
| Date:         |	December 2025 |

$$
\begin{aligned}
H_{0} &= \text{There is no statistically significant relationship between acousticness and either energy or popularity.} \\
H_{1} &= \text{Tracks with high acousticness have lower energy and lower popularity.}
\end{aligned}
$$

### Import Libraries

In [1]:
import sys
import pandas as pd
import numpy as np
from pathlib import Path
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import spearmanr

### Project Paths

In [2]:
PROJECT_ROOT = Path.cwd().parent
sys.path.append(str(PROJECT_ROOT))
DATA_DIR = PROJECT_ROOT / "data" / "clean"

INPUT_CSV = DATA_DIR / "spotify_clean.csv"

### Load Custom Libraries

In [3]:
from utils.data_processing import load_data
from utils.visualisation import plot_scatter

### Load Dataset

In [4]:
df = load_data(INPUT_CSV)
df.head()

artists                                   name  \
0        sam smith;kim petras                                 unholy   
1            bizarrap;quevedo  quevedo: bzrp music sessions, vol. 52   
2               manuel turizo                             la bachata   
3     david guetta;bebe rexha                               i'm good   
4  bad bunny;chencho corleone                        me porto bonito   

   popularity  duration_ms  explicit  danceability  energy  key  loudness  \
0         100       156943     False         0.714   0.472    2    -7.375   
1          99       198937     False         0.621   0.782    2    -5.548   
2          98       162637     False         0.835   0.679    7    -5.329   
3          98       175238      True         0.561   0.965    7    -3.673   
4          97       178567      True         0.911   0.712    1    -5.105   

   mode  speechiness  acousticness  instrumentalness  liveness  valence  \
0     1       0.0864       0.01300          0.000005    0.2660    0.238   
1     1       0.0440       0.01250          0.033000    0.2300    0.550   
2     0       0.0364       0.58300          0.000002    0.2180    0.850   
3     0       0.0343       0.00383          0.000007    0.3710    0.304   
4     0       0.0817       0.09010          0.000027    0.0933    0.425   

     tempo  time_signature      genre artist_primary  
0  131.121               4        pop      sam smith  
1  128.033               4    hip-hop       bizarrap  
2  124.980               4  reggaeton  manuel turizo  
3  128.040               4        edm   david guetta  
4   92.005               4     reggae      bad bunny

### Take Random sample

In [5]:
df_sample = df.sample(n=1000, random_state=42)

### Statistical Analysis
#### Spearman Correlation

In [6]:
r1, p1 = spearmanr(df["acousticness"], df["energy"])
r2, p2 = spearmanr(df["acousticness"], df["popularity"])

print(f"Acousticness vs Energy: ρ={r1:.3f}, p={p1:.4f}")
print(f"Acousticness vs Popularity: ρ={r2:.3f}, p={p2:.4f}")

Acousticness vs Energy: ρ=-0.715, p=0.0000
Acousticness vs Popularity: ρ=0.020, p=0.0000


##### What this means

Acousticness vs Energy
- The correlation is strong and negative.
- This indicates:
    - As acousticness increases, energy decreases substantially.
    -This aligns perfectly with musical intuition (acoustic tracks tend to be calmer).


##### Acousticness vs Popularity

- The correlation is extremely weak (close to zero).
- Despite statistical significance:
    - The effect size is negligible.
    - Popularity is not meaningfully related to acousticness alone.

#### Scatter Plot

In [7]:
fig = plot_scatter(
    df_sample,
    x_col="energy",
    y_col="acousticness",
    title="Energy vs Acousticness with Regression Line",
    xlabel="Energy",
    ylabel="Acousticness",
    trend=True
)

fig.show()

In [8]:
fig = plot_scatter(
    df_sample,
    x_col="popularity",
    y_col="acousticness",
    title="popularity vs Acousticness with Regression Line",
    xlabel="Popularity",
    ylabel="Acousticness",
    trend=True
)
fig.write_image(f"{PROJECT_ROOT}/figures/energy_vs_acousticness.png")
fig.show()

In [8]:
fig = plot_scatter(
    df_sample,
    x_col="popularity",
    y_col="acousticness",
    title="Popularity vs Acousticness with Regression Line",
    xlabel="Popularity",
    ylabel="Acousticness",
    trend=True
)
fig.write_image(f"{PROJECT_ROOT}/figures/popularity_vs_acousticness.png")
fig.show()

### 🧪 Hypothesis Decision

#### Null Hypothesis ($H_{0}$)

There is no statistically significant relationship between acousticness and either energy or popularity.

#### Decision

- Energy: ❌ Reject H₀ (strong inverse relationship)
- Popularity: ⚠️ Fail to reject H₀ in practical terms

So:
- Hypothesis 3 is partially supported.

### 🧠 Conclusion:

Hypothesis 3 is partially supported. Acousticness shows a strong and statistically significant negative relationship with energy (ρ = −0.715, p < 0.001), confirming that more acoustic tracks tend to be lower in energy. However, the relationship between acousticness and popularity is negligible (ρ = 0.020), indicating that acoustic qualities alone do not meaningfully influence a track’s popularity. This suggests that musical style affects track characteristics but does not directly translate into commercial success.